apps/web-desmos/src/screens/proposal_details/components/overview/index.tsx

```diff
@@ -12,28 +12,45 @@ import SingleProposal from '@/components/single_proposal';
 import { useProfileRecoil } from '@/recoil/profiles/hooks';
 import { readDate, readTimeFormat } from '@/recoil/settings';
 import CommunityPoolSpend from '@/screens/proposal_details/components/overview/components/community_pool_spend';
+import UpdateParams from '@/screens/proposal_details/components/overview/components/update_params';
 import ParamsChange from '@/screens/proposal_details/components/overview/components/params_change';
 import SoftwareUpgrade from '@/screens/proposal_details/components/overview/components/software_upgrade';
-import useStyles from '@/screens/proposal_details/components/overview/styles';
 import type { OverviewType } from '@/screens/proposal_details/types';
 import { getProposalType } from '@/screens/proposal_details/utils';
 import dayjs, { formatDayJs } from '@/utils/dayjs';
 import { formatNumber, formatToken } from '@/utils/format_token';
+import useStyles from './styles';
 
 const Overview: FC<{ className?: string; overview: OverviewType }> = ({ className, overview }) => {
   const dateFormat = useRecoilValue(readDate);
   const timeFormat = useRecoilValue(readTimeFormat);
   const { classes, cx } = useStyles();
   const { t } = useAppTranslation('proposals');
 
-  const types: string[] = [];
-  if (Array.isArray(overview.content)) {
-    overview.content.forEach((type: string) => {
-      types.push(getProposalType(R.pathOr('', ['@type'], type)));
-    });
-  } else {
-    types.push(getProposalType(R.pathOr('', ['@type'], overview.content)));
-  }
+  const types = useMemo(() => {
+    if (Array.isArray(overview.content)) {
+      const typeArray: string[] = [];
```

Comment:
Not important, but can avoid this `const arr = []; foo.forEach(...); return arr;` by using `foo.map(...)`, that would create a new array